### DISCUSSION TITLE
Termination status of pod is not shown

### DISCUSSION THREAD

**Discussion by gunpuz** [kubernetes-client/csharp#1359] [Q&A] [answered]
**Describe the bug**

When pod is being deleted and in kubernetes dashboard it shows that Terminating state, then `ListNamespacedPodAsync` does not represent this state
![image](https://github.com/kubernetes-client/csharp/assets/39419120/7f7e38cd-28a9-49b5-94f8-954ccbc92884)

Pod Status is not Terminating and its not clear where this terminating state is shown in returned data structure from kubernetes c# client
![image](https://github.com/kubernetes-client/csharp/assets/39419120/082e5f4e-4574-4fe2-8ce4-3470ff094702)

**Kubernetes C# SDK Client Version**
e.g. `11.0.44`

**Server Kubernetes Version**
e.g. `1.26.6`

**Dotnet Runtime Version**
e.g. net7

**To Reproduce**
Delete pod from kubernetes dashboard

**Expected behavior**
Some indication in returned data, that pod is terminating

**Comment by tg123** [comment]
see <https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase>

Note: When a Pod is being deleted, it is shown as Terminating by some kubectl commands. This Terminating status is not one of the Pod phases. A Pod is granted a term to terminate gracefully, which defaults to 30 seconds. You can use the flag --force to terminate a Pod by force.

**Reply by gunpuz** [reply]
Ok, but the main goal is not to force delete the pod, its to see its status - that it is Terminating. Kubernetes dashboard and kubectl cmd utility can extract such info. Would it be possible for kuberenetes c# library to achieve this @tg123 ?

**Reply by gunpuz** [reply]
The goal is to list pods via `ListNamespacedPodAsync` and to see the correct status of them.

**Reply by tg123** [reply]
you will see `Terminating` if delete timestamp is set in kubectl
the sdk is raw api, you can do the same as what kubectl does

**Reply by gunpuz** [reply]
I did not see this info in API, that's why this issue was created, but maybe you can guide me where to look.

**Reply by tg123** [reply]
here is why you see terminating

https://github.com/kubernetes/kubernetes/blob/d3b7391dc2f1040083ee2a8bfcb02edf7b0ded4b/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L811

**Reply by gunpuz** [reply]
@tg123 I mean in this kubernetes library, because via Kubernetes dashboard there clearly indication that pod is in process of deletion.

### This library:

I serialized `k8s.Models.V1Pod`. There is `deletionTimestamp`, but no sign of any Termination state / status....
![image](https://github.com/kubernetes-client/csharp/assets/39419120/1d5aeff9-79f4-4cb4-ba2c-46ab130d0f0d)

### Kubernetes dashboard (web):

It uses this request: 
![image](https://github.com/kubernetes-client/csharp/assets/39419120/c67924b2-05dd-4ecc-a92e-d221e1c5d2da)

![image](https://github.com/kubernetes-client/csharp/assets/39419120/e73df357-4a96-402b-8969-df738b09d83d)

**Reply by tg123** [reply, marked as answer]
Terminating is not a real state
sdk will NOT support it

some client shows status = Terminating when pod deletetimestap is set

you can do the same in your own code

**Reply by gunpuz** [reply]
Ok, thanks for the support. It would be better if there was some aggregated "state / status", but if there is not, then I could make use of `deletionTimestamp` for now. Thanks 👍!

**Reply by tg123** [reply]
you can check kubeui's impl
https://github.com/IvanJosipovic/KubeUI/commit/bef4cdbcefdf5497105bb13e4a7023ff2a6a0ad3
